content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,228 @@
+- The istio-cni node agent hops into the pod’s network namespace and establishes network redirection rules inside the pod
+network namespace, such that packets entering and leaving the pod are intercepted and transparently redirected to local
+proxy listening [ports](https://github.com/istio/ztunnel/blob/master/ARCHITECTURE.md#ports) (15008, 15006, 15001).
+- The istio-cni node agent then informs the node ztunnel over a Unix domain socket that it should establish local proxy
+listening ports inside the pod’s network namespace, (on 15008, 15006, and 15001), and provides ztunnel with a low-level
+Linux [file descriptor](https://en.wikipedia.org/wiki/File_descriptor) representing the pod’s network namespace.
+  - While typically sockets are created within a Linux network namespace by the process actually running inside that
+network namespace, it is perfectly possible leverage Linux’s low-level socket API to allow a process running in one
+network namespace to create listening sockets in another network namespace, assuming the target network namespace is known
+at creation time.
+- The node-local ztunnel internally spins up a new proxy instance and listen port set, dedicated to the newly-enrolled pod.
+- Once the in-pod redirect rules are in place and the ztunnel has established the listen ports, the pod is enrolled in the
+mesh and traffic begins flowing thru the node-local ztunnel, as before.
+
+Here’s a basic diagram showing the pod enrollment flow:
+
+{{< image width="100%"
+    link="./pod-enrollment.png"
+    alt="pod enrollment flow"
+    >}}
+
+Once the enrollment is completed, traffic to and from pods in the mesh will be fully encrypted by default, as always with Istio.
+
+Traffic will now enter and leave the pod network namespace as encrypted traffic - it will look like every ambient-enrolled
+pod has the ability to enforce mesh policy and securely encrypt traffic, even though the user application running in the pod
+has no awareness of either.
+
+Here’s a diagram to illustrate how encrypted traffic flows between meshed pods in the current ambient model:
+
+{{< image width="100%"
+    link="./traffic-flows-between-mesh-pods.png"
+    alt="traffic flows between meshed pods"
+    >}}
+
+And, as before, unencrypted plaintext traffic from outside the mesh can still be handled and policy enforced, for use cases
+where that is necessary:
+
+{{< image width="100%"
+    link="./traffic-flows-plaintext.png"
+    alt="traffic flows between meshed pods"
+    >}}
+
+### Istio Ambient Traffic Redirection: What This Gets Us
+
+The end result of the new ambient capture model is that all traffic capture and redirection happens inside the pod’s network
+namespace. To the node, the CNI, and everything else, it looks like there is a sidecar proxy inside the pod, even though
+there is, as before, **no sidecar proxy running in the pod** at all. Remember that the job of CNI implementations is to get
```

Comment:
I'd either write "there's" or something like: "it looks like there is a sidecar proxy inside the pod, even though there's **no sidecar proxy running in the pod** at all"

One the first read it sounded like you're saying "it looks like there's a sidecar car, even though there is" :D

Reply:
updated